include/uart.h

```diff
@@ -184,6 +310,211 @@ struct uart_driver_api {
+}
+
+/**
+ * @brief Set UART configuration parameter.
+ *
+ * This function should not be used directly,
+ * use functions with uart_set_ prefix instead.
+ *
+ * @param dev     UART device structure.
+ * @param setting Parameter to set.
+ * @param val     Value to set.
+ */
+_syscall int uart_configure(struct device *dev, enum uart_config setting,
+			    u32_t val);
+
+static inline int _impl_uart_configure(struct device *dev,
+				       enum uart_config setting,
+				       u32_t val)
+{
+	const struct uart_driver_api *api =
+		(const struct uart_driver_api *)dev->driver_api;
+
+	return api->configure(dev, setting, val);
+}
+
+/**
+ * @brief Set baudrate.
+ *
+ * @param dev      UART device structure.
+ * @param baudrate Baudrate.
+ *
+ * @retval -ENOTSUP Selected baudrate is not supported by device.
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_set_baudrate(struct device *dev, u32_t baudrate)
+{
+	return uart_configure(dev, UART_CFG_BAUDRATE, baudrate);
+}
+
+/**
+ * @brief Set parity.
+ *
+ * @param dev    UART device structure.
+ * @param parity Parity
+ *
+ * @retval -ENOTSUP Selected parity type is not supported by device.
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_set_parity(struct device *dev,
+				  enum uart_config_parity parity)
+{
+	return uart_configure(dev, UART_CFG_PARITY, (u32_t) parity);
+}
+
+/**
+ * @brief Set stop bits.
+ *
+ * @param dev       UART device structure.
+ * @param stop_bits Stop bits.
+ *
+ * @retval -ENOTSUP Selected number of stop bits is not supported by device.
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_set_stop_bits(struct device *dev,
+				     enum uart_config_stop_bits stop_bits)
+{
+	return uart_configure(dev, UART_CFG_STOP_BITS, (u32_t) stop_bits);
+}
+
+/**
+ * @brief Set baudrate.
+ *
+ * @param dev       UART device structure.
+ * @param data_bits Baudrate.
+ *
+ * @retval -ENOTSUP Selected number of data bits is not supported by device.
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_set_data_bits(struct device *dev,
+				     enum uart_config_data_bits data_bits)
+{
+	return uart_configure(dev, UART_CFG_DATA_BITS, (u32_t) data_bits);
+}
+
+/**
+ * @brief Set flow control.
+ *
+ * @param dev       UART device structure.
+ * @param flow_ctrl Flow control.
+ *
+ * @retval -ENOTSUP Selected flow control mode is not supported by device.
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_set_flow_control(struct device *dev,
+					enum uart_config_flow_control flow_ctrl)
+{
+	return uart_configure(dev, UART_CFG_FLOW_CONTROL, (u32_t) flow_ctrl);
+}
+
+/**
+ * @brief Set timeout for uart_rx function in blocking mode.
+ *
+ * @param dev     UART device structure.
+ * @param timeout Timeout in ms. 0 is infinite.
+ *
+ * @retval -ENOTSUP Device supports only blocking operation.
```

Comment:
Do you means 'return -ENOTSUP Device supports only non-blocking operation' ?

Reply:
Copy-paste error, removed that line